magnolia-voting-app/src/main/java/com/example/votingapp/PollStatusValueProvider.java

```diff
@@ -0,0 +1,39 @@
+package com.example.votingapp;
+
+import java.util.Calendar;
+
+import javax.jcr.*;
+
+import com.vaadin.data.ValueProvider;
+
+/**
+ * PollStatusColumnDefinition
+ */
+public class PollStatusValueProvider implements ValueProvider<Node, String> {
+    
+    @Override
+    public String apply(Node source) {
+        try {
+            if (source.hasProperties()) {
```

Comment:
This kind of logic can be reused over poll projects. So for the first step, I recommend to have an until class for handling those specific check e.g: 
DateUtils 
or seeking for an already implemented like org.apache.commons.lang.time.DateUtils

Reply:
I think checking source.hasProperties() is not clear, you should use explicitly like source.hasProperties("startDate") and source.hasProperties("expiryDate")

Reply:
@ThuyTo We do not need this check block any more.
llamkhan: I have created a public static method for handling those logic statements and supporting the purpose of reuse.